05-http-server-streams/02-file-server-post/server.js

```diff
@@ -1,22 +1,93 @@
 const url = require('url');
 const http = require('http');
 const path = require('path');
+const fs = require('fs');
+const fsPromises = require('fs/promises');
+const stream = require('stream');
+// const util = require('util');
+const LimitSizeStream = require('./LimitSizeStream');
 
+// const pipeline = util.promisify(stream.pipeline);
 const server = new http.Server();
 
-server.on('request', (req, res) => {
+server.on('request', async (req, res) => {
   const pathname = url.parse(req.url).pathname.slice(1);
-
   const filepath = path.join(__dirname, 'files', pathname);
 
   switch (req.method) {
-    case 'POST':
+    case 'POST': {
+      await fsPromises.mkdir(path.join(__dirname, 'files'), {recursive: true});
+
+      if (pathname.includes('/')) {
+        res.statusCode = 400;
+        res.end('Subfolders are not supported');
+        return;
+      }
+
+      const writeStream = fs.createWriteStream(filepath, {flags: 'wx'});
+      const limitedStream = new LimitSizeStream({limit: 1000000});
+
+      // OPTION 1 - pass tests
+      writeStream.on('error', (err) => {
+        req.unpipe(limitedStream).unpipe(writeStream);
```

Comment:
is it necessary to unpipe streams in such cases?

Reply:
I think it does not necessary. Anyway data transfer will stop